Date: Fri, 19 Jul 2019 11:39:50 -0400
From: Jarod Wilson <jarod@redhat.com>
To: rhkernel-list@redhat.com
Cc: Jarod Wilson <jarod@redhat.com>
Subject: [RHEL8 PATCH 23/45] iwlwifi: mvm: change TLC config cmd sent by rs
 to be async

Bugzilla: http://bugzilla.redhat.com/1728992

commit 5f4d55d5791a8b7150dbaba239e92719ae0f94d4
Author: Lior Cohen <lior2.cohen@intel.com>
Date:   Wed May 29 16:39:55 2019 +0300

    iwlwifi: mvm: change TLC config cmd sent by rs to be async

    The TLC_MNG_CONFIG sync cmd sent by the rs leads to a kernel warning
    of sleeping while in rcu read-side critical section. The fix is to
    change the command to be ASYNC (not blocking for the response anymore).

    Signed-off-by: Lior Cohen <lior2.cohen@intel.com>
    Signed-off-by: Luca Coelho <luciano.coelho@intel.com>
    Signed-off-by: Kalle Valo <kvalo@codeaurora.org>

Signed-off-by: Jarod Wilson <jarod@redhat.com>
---
 drivers/net/wireless/intel/iwlwifi/mvm/rs-fw.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/net/wireless/intel/iwlwifi/mvm/rs-fw.c b/drivers/net/wireless/intel/iwlwifi/mvm/rs-fw.c
index 659e21b2d4e7..be62f499c595 100644
--- a/drivers/net/wireless/intel/iwlwifi/mvm/rs-fw.c
+++ b/drivers/net/wireless/intel/iwlwifi/mvm/rs-fw.c
@@ -441,7 +441,8 @@ void rs_fw_rate_init(struct iwl_mvm *mvm, struct ieee80211_sta *sta,
 	 */
 	sta->max_amsdu_len = max_amsdu_len;
 
-	ret = iwl_mvm_send_cmd_pdu(mvm, cmd_id, 0, sizeof(cfg_cmd), &cfg_cmd);
+	ret = iwl_mvm_send_cmd_pdu(mvm, cmd_id, CMD_ASYNC, sizeof(cfg_cmd),
+				   &cfg_cmd);
 	if (ret)
 		IWL_ERR(mvm, "Failed to send rate scale config (%d)\n", ret);
 }
-- 
2.20.1


